pkgs/by-name/de/decasify/package.nix

```diff
@@ -0,0 +1,71 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+
+  # nativeBuildInputs
+  zstd,
+  pkg-config,
+  jq,
+  cargo,
+  rustc,
+  rustPlatform,
+
+  # buildInputs
+  typos,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "decasify";
+  version = "0.8.0";
+
+  src = fetchurl {
+    url = "https://github.com/alerque/decasify/releases/download/v${finalAttrs.version}/decasify-${finalAttrs.version}.tar.zst";
+    sha256 = "sha256-HTUAb/yL3H4B/n/Ecd/fDpnTYiqwco/E07sa6pFIIU4=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    nativeBuildInputs = [ zstd ];
+    dontConfigure = true;
+    hash = "sha256-GjSAdhR3TqyLP94YncFJ+XtTkrgItaupTUmIkkEZISA=";
+  };
+
+  nativeBuildInputs = [
+    zstd
```

Comment:
This would also usually go into buildInputs; is it something required for the build?

Reply:
The upstream canonical sources and ZSTD compressed, which is not quite handled automatically the way GZIP or other compressions are in Nix. See the [other comment](https://github.com/NixOS/nixpkgs/pull/355894#discussion_r1864138388) on this topic.